doc/releases/changelog-0.23.0.md

```diff
@@ -140,59 +195,118 @@
   - The quantum chemistry functionalities are unified
-    [(#2420)](https://github.com/PennyLaneAI/pennylane/pull/2420)
-    [(#2465)](https://github.com/PennyLaneAI/pennylane/pull/2465)
-    [(#2454)](https://github.com/PennyLaneAI/pennylane/pull/2454)
 
-* Adds a MERA template.
+<h4>Pattern matching optimization 🔎 💎 </h4>
+
+* Added an optimization transform that matches pieces of user-provided identity
+  templates in a circuit and replaces them with an equivalent component.
+  [(#2032)](https://github.com/PennyLaneAI/pennylane/pull/2032)
+
+  First let's consider the following circuit where we want to replace sequence of two ``pennylane.S`` gates with a
+  ``pennylane.PauliZ`` gate.
+
+  ```python
+  def circuit():
+      qml.S(wires=0)
+      qml.PauliZ(wires=0)
+      qml.S(wires=1)
+      qml.CZ(wires=[0, 1])
+      qml.S(wires=1)
+      qml.S(wires=2)
+      qml.CZ(wires=[1, 2])
+      qml.S(wires=2)
+      return qml.expval(qml.PauliX(wires=0))
+  ```
+
+  Therefore we use the following pattern that implements the identity:
+
+  ```python
+  with qml.tape.QuantumTape() as pattern:
+      qml.S(wires=0)
+      qml.S(wires=0)
+      qml.PauliZ(wires=0)
+  ```
+
+  For optimizing the circuit given the given following template of CNOTs we apply the `pattern_matching`
+  transform.
+
+  ```pycon
+  >>> dev = qml.device('default.qubit', wires=5)
+  >>> qnode = qml.QNode(circuit, dev)
+  >>> optimized_qfunc = qml.transforms.pattern_matching_optimization(pattern_tapes=[pattern])(circuit)
+  >>> optimized_qnode = qml.QNode(optimized_qfunc, dev)
+
+  >>> print(qml.draw(qnode)())
+  0: ──S──Z─╭C──────────┤  <X>
+  1: ──S────╰Z──S─╭C────┤
+  2: ──S──────────╰Z──S─┤
+
+  >>> print(qml.draw(optimized_qnode)())
+  0: ──S⁻¹─╭C────┤  <X>
+  1: ──Z───╰Z─╭C─┤
+  2: ──Z──────╰Z─┤
+  ```
+
+  For more details on using pattern matching optimization you can check the corresponding documentation and also the
+  following [paper](https://dl.acm.org/doi/full/10.1145/3498325).
+
+<h4>Measure the distance between two unitaries📏</h4>
+
+* Added the `HilbertSchmidt` and the
+  `LocalHilbertSchmidt` templates to be used for computing distance measures
+  between unitaries.
+  [(#2364)](https://github.com/PennyLaneAI/pennylane/pull/2364)
+
+  ```python
+  with qml.tape.QuantumTape(do_queue=False) as u_tape:
+      qml.Hadamard(wires=0)
+
+  def v_function(params):
+      qml.RZ(params[0], wires=1)
+
+  @qml.qnode(dev)
+  def hilbert_test(v_params, v_function, v_wires, u_tape):
+      qml.HilbertSchmidt(v_params, v_function=v_function, v_wires=v_wires, u_tape=u_tape)
+      return qml.probs(u_tape.wires + v_wires)
+
+  def cost_hst(parameters, v_function, v_wires, u_tape):
+      return (1 - hilbert_test(v_params=parameters, v_function=v_function, v_wires=v_wires, u_tape=u_tape)[0])
+  ```
+  ```pycon
+  >>> cost_hst(parameters=[0.1], v_function=v_function, v_wires=[1], u_tape=u_tape)
+  tensor(0.999, requires_grad=True)
+  ```
```

Comment:
Could be good to briefly mention what this quantity is, and to provide a link to the docstrings!